contracts/nft/src/lib.rs

```diff
@@ -0,0 +1,322 @@
+        let owner = self
+            .owner_by_id
+            .get(&token_id)
+            .expect("NonFungibleToken: token does not exist");
+        self.check_owner(owner);
+        self.check_zero_address(to);
+        self.check_approve(&token_id);
+        self.token_approvals.insert(token_id, *to);
+
+        NftEvent::Approved {
+            owner: *owner,
+            approved_account: *to,
+            token_id,
+        }
+    }
+    /// Get `ActorId` of the nft owner with `token_id`
+    fn owner(&self, token_id: TokenId) -> NftEvent {
+        let owner = self
+            .owner_by_id
+            .get(&token_id)
+            .expect("NonFungibleToken: token does not exist");
+
+        NftEvent::Owner {
+            owner: *owner,
+            token_id,
+        }
+    }
+    /// Get confirmation about approval to address `to` and `token_id`
+    fn is_approved_to(&self, to: &ActorId, token_id: TokenId) -> NftEvent {
+        if !self.owner_by_id.contains_key(&token_id) {
+            panic!("Token does not exist")
+        }
+        self.token_approvals.get(&token_id).map_or_else(
+            || NftEvent::IsApproved {
+                to: *to,
+                token_id,
+                approved: false,
+            },
+            |approval_id| NftEvent::IsApproved {
+                to: *to,
+                token_id,
+                approved: *approval_id == *to,
+            },
+        )
+    }
+
+    /// Checking the configuration with current contract data
+    fn check_config(&self) {
+        if let Some(max_mint_count) = self.config.max_mint_count {
+            if max_mint_count <= self.token_metadata_by_id.len() as u128 {
+                panic!(
+                    "Mint impossible because max minting count {} limit exceeded",
+                    max_mint_count
+                );
+            }
+        }
+    }
+    /// Check for ZERO_ID address
+    fn check_zero_address(&self, account: &ActorId) {
+        if account == &ZERO_ID {
+            panic!("NonFungibleToken: zero address");
+        }
+    }
+    /// Checks that `msg::source()` is the owner of the token with indicated `token_id`
+    fn check_owner(&self, owner: &ActorId) {
+        if owner != &msg::source() {
+            panic!("NonFungibleToken: access denied");
+        }
+    }
+    /// Checks that `msg::source()` is allowed to manage the token with indicated `token_id`
+    fn can_transfer(&self, token_id: TokenId, owner: &ActorId) {
+        if let Some(approved_accounts) = self.token_approvals.get(&token_id) {
+            if approved_accounts == &msg::source() {
+                return;
+            }
+        }
+        self.check_owner(owner);
+    }
+    /// Check the existence of a approve
+    fn check_approve(&self, token_id: &TokenId) {
+        if self.token_approvals.contains_key(token_id) {
+            panic!("Approve has already been issued");
+        }
+    }
+}
+
+#[no_mangle]
+unsafe extern fn handle() {
```

Comment:
Why `unsafe` function? Please replace to unsafe blocks.
```suggestion
extern fn handle() {
```